src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,43 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrl = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrl.add("/login");
+        allowedUrl.add("/registration");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) servletResponse;
+        HttpSession session = httpServletRequest.getSession();
+        Long driverId = (Long) session.getAttribute("driver_id");
+
+        if (driverId == null && allowedUrl.contains(httpServletRequest.getServletPath())) {
+            filterChain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+        if (driverId == null) {
+            httpServletResponse.sendRedirect(httpServletRequest.getContextPath() + "/login");
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
```

Comment:
```suggestion
        if (driverId == null || !allowedUrl.contains(httpServletRequest.getServletPath())) {
            httpServletResponse.sendRedirect(httpServletRequest.getContextPath() + "/login");
        }
        filterChain.doFilter(servletRequest, servletResponse);
```

Reply:
In last version i used wrote

 if (driverId == null && !allowedUrl.contains(req.getServletPath())) {
            resp.sendRedirect(req.getContextPath() + "/login");
        }

becouse with || authentication does not work

Reply:
Yes, you're correct in latest version, it should be &&